branding_company/model/account_invoice.py

```diff
@@ -19,16 +18,45 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp import models, fields
+from openerp import api, models, fields
 
 
 class AccountInvoice(models.Model):
     """Extend account.invoice with branding_company_id."""
     _inherit = 'account.invoice'
 
+    @api.multi
+    def onchange_partner_id(self, partner_id):
+        """When partner changes, branding company changes.
+
+        Would be nice if existing branding_company_id could be safe
+        from change, but already existing onchange method only passes
+        partner-id.
+
+        Decorater @api.onchange did not work.
+        """
+        result = super(AccountInvoice, self).onchange_partner_id(partner_id)
+        if partner_id:
+            branding_model = self.env['branding.company']
+            branding_company = (
+                branding_model.get_default_branding_company(
+                    partner_id, self.env.uid)
+            )
+            if branding_company:
+                vals = 'value' in result and result['value'] or {}
+                vals['branding_company_id'] = branding_company.id
+                result['value'] = vals
+        return result
+
+    def _get_user_branding_company(self):
+        """Default branding dependent on active user."""
+        branding_model = self.env['branding.company']
+        branding_company = (
+            branding_model.get_user_branding_company(self.env.uid))
```

Comment:
If it is part of your public API, pass the user. If only for tests
(because unittests in Odoo run as admin), set up the env and don't pass
the uid explicitely.